src/adapter/src/catalog/builtin_table_updates.rs

```diff
@@ -233,30 +233,37 @@ impl CatalogState {
         cluster_id: ClusterId,
         name: &str,
         diff: Diff,
-    ) -> BuiltinTableUpdate {
+    ) -> impl Iterator<Item = BuiltinTableUpdate> {
```

Comment:
```suggestion
    ) -> Vec<BuiltinTableUpdate> {
```

We use Vecs everywhere else for this. Let's stay with that.